### DISCUSSION TITLE
@id as string

### DISCUSSION THREAD

**Discussion by georgekrax** [prisma/prisma#10681] [Q&A] [answered]
Hello everyone,

I would like to force the **Prisma Client** to generate `string` instead of `number` for `Int` fields with @id. Is there a way to do it? I have a GraphQL API that treats `id` fields, with the corresponding string **GraphQL scalar ID** type, but TypeScript throws an error because the Client parses `@id` to `number`, and I do not want to replace `Int @id` with `String @id`. 

Thanks in advance!

**Comment by janpio** [comment, marked as answer]
> TypeScript throws an error because the Client parses @id to number, and I do not want to replace Int @id with String @id

Defining the field as a String is the way to get a String in the Client type and data.

Can you maybe cast the data to another type before using it?

**Reply by georgekrax** [reply]
@janpio do you mean implementing a custom `ID` scalar type?

**Reply by georgekrax** [reply]
I am using the `nexus` library with `nexus.prisma`. So my types are something like:
```
t.field(User.id)
```

Can we implement something similar with the `projectIntToId` setting, but for BigInts too?

**Reply by janpio** [reply]
I don't know, this is the Prisma repository so I can only tell you what Prisma does - and their you get what is defined in the Prisma Schema. Then you can do with the data whatever you want to cast it into different types and so on in your application of course.

**Reply by georgekrax** [reply]
In TypeORM for `id` I can pass a `string` or a `number`. Would be nice to have an option to include strings with numbers. I cannot always cast objects like this:

```js
user = { ...user, id: user.id.toString() }
```

**Reply by janpio** [reply]
That would be a feature request then. Feel free to open an issue and include motivation and an example waht that could look like.

**Comment by hussamkhatib** [comment]
@georgekrax 
This should do 
```prisma
id  String  @id @default(cuid())
```

**Reply by SirJosh1987** [reply]
I only have a phone atm my computer has been fired. No saving her. How would I do everything without being on a computer?

**Reply by georgekrax** [reply]
@hussamkhatib I do not use String ID in the schema. I use Int or BigInt, which are safer and easier to manipulate

**Reply by georgekrax** [reply]
But I want the Prisma Client to return IDs as strings, and not integers because it is easier to manipulate them in backend. TypeORM returns them as strings

**Comment by mqschwanda** [comment]
For @georgekrax or anyone still looking for a solution when working with GraphQL and Prisma IDs,  you can use [graphql-codegen](https://the-guild.dev/graphql/codegen/plugins/typescript/typescript#scalars) to generate your types with custom scalar mappings that accepts a string or number.

**Comment by FezVrasta** [comment]
Even with `emitIdAsIDType` the filter types are wrong and expect me to do `where: { id: { equals: String!} }` instead of just `where: { id: ID! }`. Is there any solution to this?
